explorer/data/prelude.carbon

```diff
@@ -416,6 +416,56 @@ external impl i32 as RightShiftWith(i32) where .Result == i32 {
   }
 }
 
+//-------------------------
+// Optional
+//-------------------------
+choice OptionalElement(T:! Type) {
+  None(),
+  Element(T)
+}
+
+class Optional(T:! Type){
+  fn CreateEmpty() -> Optional(T){
+    return { .element = OptionalElement(T).None() };
+  }
+  fn Create ( value: T ) -> Optional(T){
+    return { .element = OptionalElement(T).Element(value) };
+  }
+
+  fn has_value[me: Self] () -> bool{
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).None() => { return false; }
+    }
+    return true;
+  }
+
+  fn get[me: Self] () -> T {
+    var y: T;
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).Element( var x: T ) =>{
+        return x;
+      }
+    }
+    return y;
+  }
+
+  impl Optional(T) as EqWith(bool){
+    fn Equal[me: Self](other: bool) -> bool{
+      return me.has_value();
+    }
+  }
```

Comment:
I don't think we should support `==` comparisons between optional values and `bool`. Those seem a little too error-prone to me.

Reply:
It's ok, removed!